### DISCUSSION TITLE
Suggestion on working around this annoying behavior with labels?

### DISCUSSION THREAD

**Discussion by acut3** [folke/flash.nvim#203] [Q&A] [answered]
Let me start by saying that I know next to nothing about Flash. I'm using LazyVim, and only learned about Flash when I realized it was the cause of the annoying issue I was experiencing.

So, when Flash is enabled and you do a search, it will propose "labels" and jump to the corresponding location when that label is entered. The issue is, if I'm searching for a word the buffer doesn't contain, chances are I will type a key that happens to be one of the proposed labels. When it happens, Flash stops the search and jumps to the location of the label. Any additional key that is then pressed is now a normal vim command.

So, let say I'm searching for `flash` and the buffer only has matches for `f` not followed by an `l`. I will type a quick `<esc>/flash`. When `l` is typed, Flash will jump to the location of label `l` (assuming there is one). Then `a` switches to insertion mode, and `sh` will be inserted into the buffer. I just wanted to search for a word, and I've accidentally modified the buffer.

Now how anyone can find this behavior acceptable, I don't know. I was literally going crazy trying to figure out what was happening. Took me a while to realize Flash was the culprit, and I eventually disabled Flash completely.

Now if I wanted to still use Flash, how would I disable labels and keep the other functionalities?

**Comment by IndianBoy42** [comment, marked as answer]
Please expand the "> default settings" section in the readme, or see the help docs for how to configure, enable and disable any functionality. `modes.search.enabled = false` will disable it in the search prompt. There is also a toggle function that you can map to toggle the labels in search mode. There is also `label.min_pattern_length` to only show labels after a certain number of chars, so you'll reduce the chance of mistakes as long as you take a pause before going past that number of chars.

> Now how anyone can find this behavior acceptable, I don't know. I was literally going crazy trying to figure out what was happening. Took me a while to realize Flash was the culprit, and I eventually disabled Flash completely.

New behaviour, especially one that you added without fully understanding, will always be surprising. You can either adapt to it if you feel it will improve your workflow or disable it or tweak it, it doesnt necessarily make it unacceptable.

The thing with the search labels is that `/?` can be used for two similar but distinct purposes. 

1. Jumping to a certain piece of text that you know exists but maybe don't know where it is (maybe because its off screen)
2. Finding whether some piece of text exists in the buffer 

The dynamic search labels are pretty useful in the former case but detrimental in the latter case. Personally I have separate keymaps with flash configured slightly differently to cater for each of these cases.

**Comment by acut3** [comment]
Thanks, `modes.search.enabled = false` did the trick.

I get your point about new behavior, but I still think the usability of this feature is not great. It's not just annoying (to me), it's also pretty dangerous as it could lead the user to modify their buffer without even realizing it. It might not be an issue when someone install Flash willingly, but since it comes as a default with LazyVim, it *will* affect many people that didn't ask for it.

I don't know if it's technically doable, but I think it would be much better to require a combination of keys to jump to a label (like Ctrl+label), instead of automatically transitioning from "searching" to "jumping to label" without any voluntary action from the user.

**Comment by acut3** [comment]
I can't figure a way to disable just labels for normal searches (with `/`), but still keep Flash's match counts (the `[n/N]` at the end of each matching line). Is there a way of doing that?

**Comment by jgregoire** [comment]
I set ```label.min_pattern_length = 2``` to emulate Leap's behavior. I then limit my search query to 2 characters, which is generally Close Enough.
